clearly-defined/src/main/kotlin/ClearlyDefinedService.kt

```diff
@@ -239,15 +239,41 @@ interface ClearlyDefinedService {
     )
 
     /**
-     * See https://github.com/clearlydefined/service/blob/b339cb7/schemas/curations-1.0.json#L64-L83.
+     * See https://github.com/clearlydefined/service/blob/b339cb7/schemas/curations-1.0.json#L64-L83 and
+     * https://docs.clearlydefined.io/using-data#a-note-on-definition-coordinates.
      */
     @JsonInclude(JsonInclude.Include.NON_NULL)
     data class Coordinates(
-        val name: String,
-        val namespace: String? = null,
+        /**
+         * The type of the component. For example, npm, git, nuget, maven, etc. This talks about the shape of the
+         * component.
+         */
+        val type: ComponentType,
+
+        /**
+         * Where the component can be found. Examples include npmjs, mavencentral, github, nuget, etc.
+         */
         val provider: Provider,
-        val type: ComponentType
-    )
+
+        /**
+         * Many component systems have namespaces: GitHub orgs, NPM namespace, Maven group id, etc. This segment must be
+         * supplied. If your component does not have a namespace, use '-' (ASCII hyphen).
```

Comment:
If the namespace must be supplied, why is it nullable?

Reply:
I guess  the (original) comment is misleading: In the data structure the namespace does not have to be provided, but in any user-facing (string) representation "-" should be used instead of nothing / an empty string.